backend/tomato/host/__init__.py

```diff
@@ -175,27 +177,34 @@ def incrementErrors(self):
 	def update(self):
 		self.availability *= config.HOST_AVAILABILITY_FACTOR
 		self.save()
+		
 		if not self.enabled:
 			return
 		before = time.time()
-		self.hostInfo = self._info()
-		after = time.time()
-		self.hostInfoTimestamp = (before + after) / 2.0
-		self.hostInfo["query_time"] = after - before
-		self.hostInfo["time_diff"] = self.hostInfo["time"] - self.hostInfoTimestamp
+		
+		
 		try:
+			self.hostInfo = self._info()
+			self.synchronized = True
+		
+			after = time.time()
+			self.hostInfoTimestamp = (before + after) / 2.0
+			self.hostInfo["query_time"] = after - before
+			self.hostInfo["time_diff"] = self.hostInfo["time"] - self.hostInfoTimestamp
 			self.hostNetworks = self.getProxy().host_networks()
+			caps = self._convertCapabilities(self.getProxy().host_capabilities())
+			self.elementTypes = caps["elements"]
+			self.connectionTypes = caps["connections"]
+			self.componentErrors = max(0, self.componentErrors / 2)
+			if not self.problems():
+				self.availability += 1.0 - config.HOST_AVAILABILITY_FACTOR
+			self.save()
+			logging.logMessage("info", category="host", name=self.name, info=self.hostInfo)
+			logging.logMessage("capabilities", category="host", name=self.name, capabilities=caps)
 		except:
-			self.hostNetworks = []
-		caps = self._convertCapabilities(self.getProxy().host_capabilities())
-		self.elementTypes = caps["elements"]
-		self.connectionTypes = caps["connections"]
-		self.componentErrors = max(0, self.componentErrors / 2)
-		if not self.problems():
-			self.availability += 1.0 - config.HOST_AVAILABILITY_FACTOR
-		self.save()
-		logging.logMessage("info", category="host", name=self.name, info=self.hostInfo)
-		logging.logMessage("capabilities", category="host", name=self.name, capabilities=caps)
+			self.synchronized = False
```

Comment:
You should raise the exception after setting the flag to keep the original semantics.